### ISSUE TITLE
"Error: Not Found" if push event's `before` commit is `0000000000000000000000000000000000000000`

### ISSUE THREAD

**Issue by douglascayers** [pmd/pmd-github-action#128]
### Scenario

When configured to run on a `push` event and the GitHub's `context.payload.before` commit is `0000000000000000000000000000000000000000` (meaning there was no prior commit) then the action fails with error `Error: Not Found`.

The error is raised by `util.js` by `determineModifiedFiles(..)` function when it invokes `octokit.rest.repos.compareCommitsWithBasehead(..)`

https://github.com/pmd/pmd-github-action/blob/main/lib/util.js#L143

### Config File

```yaml
# .github/workflows/pmd.yml
name: Run PMD

on:
    push:
        branches:
            - my-branch
        paths:
            - my-folder

jobs:
    build:
        name: Run PMD
        runs-on: self-hosted
        steps:
            - name: Checkout repository
              uses: actions/checkout@v3

            - name: Run Apex Rules
              uses: pmd/pmd-github-action@v1
              with:
                  rulesets: rulesets/PMD/apex-rules-github.xml
                  sourcePath: my-folder
                  analyzeModifiedFilesOnly: true
                  createGitHubAnnotations: true
```

### Expected Result

* PMD to not perform a diff comparison because there is no prior commit.
* PMD to identify the list of files included in the commit for the given source path.

[Example](https://stackoverflow.com/a/424142/470818):
```shell
$ git diff-tree --no-commit-id --name-only -r <commit>
index.html
my-folder/some-file.java
```

### Actual Result

* Action fails with the following error

```
##[debug]finished caching tool
Using PMD 6.49.0 from cached path /home/docker/actions-runner/_work/_tool/pmd/6.49.0/x64
Determining modified files in my-folder...
##[debug]Push on refs/heads/my-branch: 0000000000000000000000000000000000000000...c46068a4fd0827f6c4365d24c02c1bc44707287a
Error: Not Found
##[debug]Node Action run completed with exit code 1
```

### Workaround

* Push another commit to the branch
